src/s2/s2region_term_indexer_test.cc

```diff
@@ -131,6 +134,8 @@ TEST(S2RegionTermIndexer, IndexRegionsQueryPointsOptimizeTime) {
   options.set_max_level(16);
   options.set_max_cells(20);
   TestRandomCaps(options, QueryType::POINT);
+  options.set_query_contains_points_only(true);
```

Comment:
I think we can remove `QueryType` and do it based only on `query_contains_points_only`.  This is similar to how it works for `index_contains_points_only`.

@smcallis what do you think?

Reply:
> I think we can remove QueryType and do it based only on query_contains_points_only. This is similar to how it works for index_contains_points_only.

No. We want to test which query points with and without points only optimization

Reply:
Why? Points are also regions, and this property is already used here: https://github.com/google/s2geometry/pull/303/files/0f9ef7a0ebe8d52ebf73cfb4c441cc1e3d553a05#diff-c9f79443ff2537a812dd6c149afd00960afcc88655f0514de8a1cc4741e78e79L66

I agree it would be more test coverage, but I'm fine making this same assumption we're already making.

Reply:
In general exist such scenarios matrix:

index contains only points, only points with enabled optimization, anything
x
same about query 
x
time/space bool option


So 3*3*2 = 18 configuration

Do you agree?

Reply:
Ok I understood what do you mean, you are incorrect.

Now in master tested such cases:

index contains points: true/false
query points/query regions

index contains points optimization affect query terms!

So for opposite options I need to make same query (points only) with two different index

Reply:
Anyway I wrote better coverage with automatic case combinations

Reply:
`16 (2 * 2 * 2 * 2) - 4 (2 * 2 * 1 * 1) = 12`
all combinations - skipped
as expected

Reply:
Hmm you right about now in my code wasn't checked query points but index contains caps.

Fixed.

`2 * 2 * 2 * 2 * 2 - 2 * (1 * 2 * 2 * 1 * 2) = 16`

cases now

Reply:
@jmr Do you ok with current tests?